weave/integrations/langchain/langchain.py

```diff
@@ -72,38 +75,119 @@ def _persist_run(self, run: Run) -> None:
+            `RunnableSequence1` popped onto the stack. To solve for this, we need to send
+            `False` as the `parent_id`, telling the system: "trust me, this is a root".
+            """
+            parent_run: Optional[Call] = None
+            lc_parent_run_id = (
+                str(run.parent_run_id) if run.parent_run_id is not None else None
+            )
+            wv_parent_run = (
+                self._call_map.get(lc_parent_run_id) if lc_parent_run_id else None
+            )
 
-            # Create a call object.
-            is_valid_root = run.parent_run_id is None
-            is_valid_child = run.parent_run_id in self._call_map
-
-            if is_valid_root or is_valid_child:
-                # TO:DO, Figure out how to handle the run name. It errors in the UI
-                run_name = make_valid_run_name(run.name)
-                parent_id = run.parent_run_id
-                call = self.gc.create_call(
-                    # Make sure to add the run name once the UI issue is figured out
-                    f"langchain.{run.run_type.capitalize()}.{run_name}",
-                    self._call_map.get(parent_id),
-                    run_dict,
-                )
+            create_call_fn = self.gc.create_call
+            if wv_parent_run is not None:
+                parent_run = wv_parent_run
+            else:
+                # Here is our check for the specific condition
+                wv_current_run = run_context.get_current_run()
+
+                # First, there needs to be something on the stack.
+                if wv_current_run is not None:
+                    attrs = wv_current_run.attributes or {}
+                    # Now, the major condition:
+                    if (
+                        # 1. Both runs must be of type `RunnableSequence`
+                        attrs.get("lc_name") == run.name == RUNNABLE_SEQUENCE_NAME
+                        # 2. Both us and the sibling run must have empty parents (i think
+                        # this condition will always be true, else we would have a parent
+                        # run already, but trying to be safe here)
+                        and attrs.get("parent_run_id") == lc_parent_run_id == None
+                    ):
+                        # Now, we know that Langchain has confused us. And we want to set the
+                        # parent to the current Weave Run's parent. So, if that parent is
+                        # None, then we use `False` here to force a root. Else we lookup
+                        # the parent from the client. You might be thinking... id the parent
+                        # run_id is none, when would this NOT be none? The answer: when Langchain
+                        # is called inside of another op (like Evaluations!)
+                        if wv_current_run.parent_id is None:
+                            create_call_fn = self.gc.create_call_outside_execution
```

Comment:
given the new api, instead of switching the call fn, just pass `use_stack=False` to `create_call` now